### ISSUE TITLE
Resources in Search page show up as 'undefined'

### ISSUE THREAD

**Issue by veuncent** [archesproject/arches#1641]
#### Current Behavior
After I saved a new resource and navigated to the Search page, my new resource shows up as 'undefined':

![image](https://cloud.githubusercontent.com/assets/7893339/23041266/4a2ba7a8-f462-11e6-97e2-72652a043792.png)

The result is the same for authenticated and unauthenticated users.

#### Expected Behavior
I expected the resource name to be displayed instead of the first occurrence of 'undefined'.

#### Steps to Reproduce (for bugs)

1. Open form to create a new "Heritage Resource Model"
2. Enter a Name
3. Select 'Primary' from "Name Type"
4. Click Save
5. Navigate to Search page

#### Your Arches Information

* Version used: Master branch, commit SHA: 36d1e7c3cc0ca0d6564ae67b14c00efe557d5953
* Browser Name and version: Firefox 51.0.1 (32-bit) & Chrome 56.0.2924.87
* Operating System and version (desktop or mobile): Windows 10 desktop

#### Context
Arches is running on localhost in a Docker container. This might not necessarily affect this bug, but an incorrect installation of Arches could be a cause.

**Comment by dwuthrich** [comment]
@veuncent This issue exists when a user creates a resource instance without first configuring the "Define Resource Descriptors" function.  

Because Arches does not assume anything about how you want to identify a resource instance, we allow you to configure the identifier and content that will be displayed as part of search with the Define Resource Descriptors function.

Steps to resolve:
1. From Arches Designer, navigate to functions for the resource model that you wish to configure
2. Select Define Resource Descriptors and configure the function

**Comment by veuncent** [comment]
Got it! Thanks for explaining. 

Will there be a Getting Started guide that explains steps like this? I couldn't find it in the Github wiki. 
If so, is there a draft version I might see? I am trying to get a head start with my resource graphs before Arches v4 goes life. I'm also working on Dockerizing the app.

**Comment by dwuthrich** [comment]
Yes, we're working on documenting Arches.  @mradamcox is leading the effort and he may be able to summarize how you can review the draft documentation.   I'll assume that you will close this issue after discussing with him.

Also great to hear that you're Dockerizing the application!

**Comment by veuncent** [comment]
Awesome, thanks! Closing this (non-)issue...

**Comment by mradamcox** [comment]
Hi @veuncent , the documentation I have drafted for functions is not quite in a pull request yet, but you can preview it by switching to the `1548_designer_help` branch. Or, easier, you could log into my own test installation (which is currently on that branch) and head to the function designer as @dwuthrich mentioned. arches4.legiongis.com use admin as username and password to log in. Once on the Function Designer page click the ? button for help.

This isn't an overall "getting started" guide yet, but is the more detailed documentation for that part of the app.

Any feedback is appreciated!

**Comment by veuncent** [comment]
Awesome, thank you Adam. 
It's great to have specific help sections directly on the relevant page. 
Also, configuring Arches like this seems to work quite well. Great to have freedom. 

Feedback #1:  
Just a thought, but have you considered displaying the Help button ("?") in a more prominent place on the page, in addition to the top right? Possibly users could overlook that button.

Feedback #2:  
To me personally the vocabulary used in the Help section and more broadly throughout the Arches project is quite hard to follow. I usually have to read paragraphs a couple of times before I figure out the meaning of the jargon. Now I do not consider my own vocabulary representative of the Arches community, but I do think my data entry users (colleagues) will have trouble understanding, because they have less technical database knowledge. For us this is less of a problem, because I plan to take most responsibility for the database other than data entry (and I will just read the paragraphs three times ;) ), but for other Arches user teams it might be an issue. 


Question:  
Is it currently possible to set Display Name according to a specific Name Type? E.g. always display  the <Name> value where Name Type = Secondary Name

**Comment by mradamcox** [comment]
Thanks a bunch for the feedback. We may be able to make the help button stand out a little more (maybe change "?" to "Help"), or perhaps we'll can alert people to its existence right away so they know to look for it. Worth discussion for sure.

As for vocabulary, I'll say 1. that sadly I tend to the "too much" end of the verbosity spectrum, especially on first drafts. Hopefully some basic editing will address that at least in part. 2. Yes, there is a lot of terminology! In many ways I don't think we'll be able to get around that, because this "Arches Designer" part of the app is pretty complex. On the other hand, I hope to add some glossary-type diagrams that will help people see how everything fits together.

Having read through some of these help menus, are there specific terms/concepts that are still mysterious to you? Would love to know... and thanks for looking!

**Comment by mradamcox** [comment]
Hi @veuncent , I have added a good deal of new documentation that is in the help panel for the Arches Designer landing page (the one that lists all of the Branches and Resource Models). You can see it in the arches instance I linked to above. I added definitions for terminology, and I'd like to know what you think of it.

**Comment by veuncent** [comment]
Hey @mradamcox , I would be honored to be your guinee pig. I will find time over the next couple of days. Stay tuned

**Comment by mradamcox** [comment]
@veuncent Thanks, you can get in touch at adam @ legiongis.com.

**Comment by veuncent** [comment]
After completing these two steps, my resources still show up as 'undefined' in the search results:

- From Arches Designer, navigate to functions for the resource model that you wish to configure
- Select Define Resource Descriptors and configure the function

Is this supposed to work yet or am I just too early to the party?

Thanks!

**Comment by veuncent** [comment]
Fixed:

The function is run only when the resource is indexed (which happens on the "save" operation). So if you create a bunch of resources, and then add the function, they will still read "undefined" until they are indexed.
